### ISSUE TITLE
max size of object limited to 75% of image

### ISSUE THREAD

**Issue by MasIgor** [ultralytics/ultralytics#17718]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi all!

I am facing a problem with my yoloV8 network.
The max height of a rectangle is 75% of the image. if the object is larger, then thebox is limited to 75% height and the score lowered. if the size is bigger then all is well. 

Both images are taken from the training dataset and the training itself reaches pretty good values:
P: 0.999
R: 1
MAP0.5: 0.995
MAP95: 0.968

Why is this happening and how can I unlimit the height of the rectangle?
any help appreciated!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @MasIgor, thank you for bringing this to our attention and for your interest in Ultralytics 🚀! While you're waiting for an Ultralytics engineer to take a closer look, here are a few things you might find useful.

If this is a 🐛 Bug Report and you haven't already done so, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug the issue. This would help in understanding the context and possibly identifying what's limiting the height of the detection box.

In the meantime, we recommend checking out the [Docs](https://docs.ultralytics.com) for detailed guidance. You might find useful information in sections related to [Python](https://docs.ultralytics.com/usage/python/) or [CLI](https://docs.ultralytics.com/usage/cli/) that could aid you in troubleshooting and potentially resolving your issue.

For real-time support and community interaction, consider joining the Ultralytics community that best suits you. Engage in conversations and get tips on [Discord](https://ultralytics.com/discord) 🎧, participate in discussions on [Discourse](https://community.ultralytics.com), or explore our [Subreddit](https://reddit.com/r/ultralytics) for more insights.

## Upgrade

Ensure that you're running the latest version of the `ultralytics` package along with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). An update might resolve your issue:

```bash
pip install -U ultralytics
```

## Environments

You can also run YOLO in one of these verified environments, which come with all necessary dependencies:

- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud**, **Amazon**, or **Docker Image**. Relevant quickstart guides and badges are available in our documentation links.

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

This badge indicates if all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. These tests confirm the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on various operating systems. 

We appreciate your patience and look forward to assisting you further! 😊

**Comment by Y-T-G** [comment]
You can try decreasing the imgsz and use P6 model.

https://github.com/ultralytics/ultralytics/issues/16895#issuecomment-2413016050

**Comment by MasIgor** [comment]
@Y-T-G  thanks for your anser...

the images are 640x640. it worked well with Y5. i am using the S model, so to troubleshoot i tested the M model. both models present the exact same result. in 64000 images there is not a single image with a rectangle bigger than 75%.

**Comment by Y-T-G** [comment]
You can try running inference at imgsz=320

**Comment by MasIgor** [comment]
yes i can try that for a test, but cannot leave it like that as some classes are recognized by tiny features that are too small at that resolution...

**Comment by Y-T-G** [comment]
If you use a large imgsz, the receptive field won't be big enough to see the whole object. You can use P6 model like I said.

**Comment by MasIgor** [comment]
thank you, just starting a training with the yolov8s-p6.yaml. 

could you please explain what the p6 exactly is? i hear it for the first time now. thank you

**Comment by Y-T-G** [comment]
It has an extra scale for extra large objects. It increases the receptive field of the anchors.

**Comment by MasIgor** [comment]
thank you!

standard s model uses P3 correct? is there a way to train P4 instead of P6? that would keep it faster

**Comment by Y-T-G** [comment]
The default models are P3-P5

**Comment by MasIgor** [comment]
@Y-T-G  the P6 solved the problem thank you! 
what does it mean that it is P3-P5? does it use all of them? and the P6 uses all from P3 to P6?

**Comment by Y-T-G** [comment]
Yes

**Comment by MasIgor** [comment]
@Y-T-G  just one more quick question. I have 3 networks described like this:

**Network 1:**
type S, 
138 classees, 
---> rectangles only 75% of the image.

**Network 2:** 
type S-P6 as suggsted, 
138 classes, 
---> rectangles OK
 
**Network 3:** 
type S, 
36 classes 
---> rectangles OK.

all networks take pictures of the same type of object (crate). the difference is only in the amount of different classes. all images have the same size, objects in the same size, same light, all the same. 

so far, it is clear to me why 1 does not work, and why 1.1 works. but i cannot understand why network 2 works. how does the number of classes influence the size of the drawn rectangle?

**Comment by Y-T-G** [comment]
> so far, it is clear to me why 1 does not work, and why 1.1 works. but i cannot understand why network 2 works

I don't get it. You said rectangles are ok for network 2.

**Comment by MasIgor** [comment]
sorry, typo there..

so far, it is clear to me why 1 does not work, and why 2 works. but i cannot understand why network 3 works. how does the number of classes influence the size of the drawn rectangle?

**Comment by Y-T-G** [comment]
Since you have large objects, the only scales that matter are P5 and P6. The objects are assigned to the scales based on their centers. If they get assigned to P3 and P4 during training, the model will use scales that can't predict large objects. I guess the chance of this increases when you have more objects.

**Comment by MasIgor** [comment]
thank you very much! very clear

**Comment by MasIgor** [comment]
@Y-T-G Hello, first off, happy new year.. :) 

reopening again as there still are some issues..

these are the YoloV8 networks i trained..

S: as seen above, draws only 75% of the image height as max box height.
S-P6: draws correctly, but has many misrecognitions.
M: recognizes correctly but draws only 75%
L: recognizes correctly but draws only 75%.

Is there a way to train a M or L network in such a way that it forces to draw bigger rectagnles? its my understanding that the P6 level is automatically enabled for the M and L networks.. why are they not being used to recognize the object?

my images are 640x640. the object fills 95% of the image height and about 60% of the image width

any help appreciated

**Comment by Y-T-G** [comment]
No. None of the default models have P6 level, even if the size is larger
